pdr_backend/lake/table_bronze_pdr_slots.py

```diff
@@ -54,46 +54,50 @@ def get_bronze_pdr_slots_data_with_SQL(
-                filtered_{etl_bronze_pdr_predictions_table_name}.timeframe as timeframe,
-                filtered_{etl_bronze_pdr_predictions_table_name}.source as source,
-                {pdr_slots_table_name}.slot as slot,
-                filtered_{pdr_payouts_table_name}.roundSumStakesUp as roundSumStakesUp,
-                filtered_{pdr_payouts_table_name}.roundSumStakes as roundSumStakes,
-                {pdr_truevals_table_name}.truevalue as truevalue,
-                {pdr_slots_table_name}.timestamp as timestamp,
+            SELECT DISTINCT
+                {pdr_slots_table_name}.ID AS ID,
+                SPLIT_PART({pdr_slots_table_name}.ID, '-', 1) AS contract,
+                {etl_bronze_pdr_predictions_table_name}.pair AS pair,
+                {etl_bronze_pdr_predictions_table_name}.timeframe AS timeframe,
+                {etl_bronze_pdr_predictions_table_name}.source AS source,
+                {pdr_slots_table_name}.slot AS slot,
+                {pdr_payouts_table_name}.roundSumStakesUp AS roundSumStakesUp,
+                {pdr_payouts_table_name}.roundSumStakes AS roundSumStakes,
+                {pdr_truevals_table_name}.truevalue AS truevalue,
+                {pdr_slots_table_name}.timestamp AS timestamp,
                 GREATEST(
                     {pdr_slots_table_name}.timestamp,
-                    filtered_{pdr_payouts_table_name}.timestamp,
-                    filtered_{etl_bronze_pdr_predictions_table_name}.timestamp,
-                    {pdr_truevals_table_name}.timestamp
-                ) as last_event_timestamp,
+                    {pdr_truevals_table_name}.timestamp,
+                    {pdr_payouts_table_name}.timestamp,
+                    {etl_bronze_pdr_predictions_table_name}.timestamp
+                ) AS last_event_timestamp
             FROM
                 {pdr_slots_table_name}
-            LEFT JOIN {pdr_truevals_table_name}
-            ON {pdr_slots_table_name}.ID = {pdr_truevals_table_name}.ID
+            LEFT JOIN {pdr_truevals_table_name} ON {pdr_slots_table_name}.ID = {pdr_truevals_table_name}.ID
             LEFT JOIN (
-                SELECT *,
-                    SPLIT_PART({pdr_payouts_table_name}.ID, '-', 1)
-                || '-' || SPLIT_PART({pdr_payouts_table_name}.ID, '-', 2) as match_id,
-                ROW_NUMBER() OVER (PARTITION BY match_id) as rn,
-                FROM {pdr_payouts_table_name} 
-            ) AS filtered_{pdr_payouts_table_name} 
-            ON {pdr_slots_table_name}.ID = filtered_{pdr_payouts_table_name}.match_id
-            AND filtered_{pdr_payouts_table_name}.rn = 1
+                SELECT 
+                    {pdr_slots_table_name}.ID AS slot_id,
+                    MAX({etl_bronze_pdr_predictions_table_name}.pair) AS pair,
+                    MAX({etl_bronze_pdr_predictions_table_name}.timeframe) AS timeframe,
+                    MAX({etl_bronze_pdr_predictions_table_name}.source) AS source,
```

Comment:
the pair, timeframe and source columns are string values, I think using `MAX` does not make sense

Reply:
MAX() will return the string value that appears last in the collation order.

Reply:
As I wrote here... these should be retrieved from subgraph so the data is just there from the initial fetch from the subgraph... No need to execute a join or a max.
https://github.com/oceanprotocol/pdr-backend/issues/989

And... if you just need one record in SQL use `first()` as the scan/cost of the call will be cheaper or `last()`